src/Compilers/CSharp/Portable/Binder/PatternVariableFinder.cs

```diff
@@ -37,10 +44,21 @@ class PatternVariableFinder : CSharpSyntaxWalker
 
             var result = finder.declarationPatterns;
             finder.declarationPatterns = null;
+            Debug.Assert(finder.expressionsToVisit.Count == 0);
             s_poolInstance.Free(finder);
             return result;
         }
 
+        private void VisitExpressions()
+        {
+            while (expressionsToVisit.Count != 0)
```

Comment:
See how it was done for declaration expressions in LocalScopeBinder.cs.
The implementation of VisitBinaryExpression of interest was removed in https://github.com/dotnet/roslyn/commit/9d36a64a.